src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/CreateCohortController.cs

```diff
@@ -35,20 +57,55 @@ public IActionResult Index(IndexRequest request)
         [Route("select-provider")]
         public IActionResult SelectProvider(SelectProviderRequest request)
         {
-            var viewModel = new SelectProviderViewModel();//todo: from mapper
+            if (!ModelState.IsValid)
+            {
+                return RedirectToAction("Error", "Error", new { StatusCode = 400 });
+            }
+
+            var viewModel = _selectProviderViewModelMapper.Map(request);
 
             return View(viewModel);
         }
 
         [Route("select-provider")]
         [HttpPost]
-        public IActionResult SelectProvider(SelectProviderViewModel request)
+        public async Task<IActionResult> SelectProvider(SelectProviderViewModel request)
         {
-            //todo:hit api
+            if (!ModelState.IsValid)
+            {
+                return RedirectToAction("Error", "Error", new { StatusCode = 400 });
+            }
+
+            try
+            {
+                var validationResult = _selectProviderViewModelValidator.Validate(request);
```

Comment:
We don't need this explicit call - should be done automatically, and the results added to the model state so we should get them above

Reply:
done